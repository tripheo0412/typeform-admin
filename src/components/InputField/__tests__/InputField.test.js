import React from 'react';

import { shallow, mount } from 'enzyme';

import { InputField } from '..';

describe('InputField', () => {
  let wrapper;

  const props = {
    id: 1,
    type: 'email',
    placeholder: 'Your Email',
    isSignin: 'false',
    handleChange: () => {},
  };

  it('InputField should render correctly', () => {
    wrapper = shallow(<InputField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('InputField should render correctly type prop', () => {
    wrapper = mount(<InputField {...props} />);
    expect(wrapper.props().type).toBe('email');
  });
  it('InputField variant prop should be inputField--primary by default', () => {
    wrapper = mount(<InputField {...props} />);
    expect(wrapper.props().variant).toBe('inputField--primary');
  });
  it('InputField isSignin prop should be false by default', () => {
    wrapper = mount(<InputField {...props} />);
    expect(wrapper.props().isSignin).toBe('false');
  });
  it('InputField onChange function should be called', () => {
    const mockFunc = jest.fn();
    wrapper = mount(<InputField {...props} handleChange={mockFunc} />);
    wrapper.find('input').simulate('change');
    expect(mockFunc).toHaveBeenCalled();
  });
});
