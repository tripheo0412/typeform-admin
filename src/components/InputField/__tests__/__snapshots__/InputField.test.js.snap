// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField InputField should render correctly 1`] = `ShallowWrapper {}`;
